server.js

```diff
@@ -1,25 +1,55 @@
 // Require the Express module (https://npmjs.com/package/express)
-
+var express = require("express");
 
 // Create a new express application instance by calling `express()`
-
+var app = express();
 
 // Serve files in the 'public' directory with Express's built-in static file server
-
+app.use(express.static('public'));
 
 // Create a Counter class that will be used to create counter objects
 // See the full description in README.md
 
 
+function Counter() {
+    this.pet = {
+        puppies: 0,
+        kittens: 0
+    };
+}
+
+Counter.prototype.record = function(key) {
+    this.pet[key]++;
```

Comment:
The `.record` method should check if there is already property called `key` on the `.hasVotes` object. If the key *doesn't* exist, initialize it to zero.

You can either use the [`in` operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in) and say:

```js
if(key in this.haveVotes) {
  /* increment */
}
 else {
 /* initialize to 1 */
}
```

Or you can use the [`.hasOwnProperty()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty) method:

```js
if(this.haveVotes.hasOwnProperty(key)) {
  /* increment */
}
else {
 /* initialize to 1 */
}
```

If the property doesn't exist, its value will be `undefined`, which is a “falsy” value. So you could also just do:


```js
if(this.haveVotes[key]) {
  /* increment */
}
else {
 /* initialize to 1 */
}
```

Any of these approaches would work fine.